Source/WebCore/rendering/RenderListMarker.cpp

```diff
@@ -2025,9 +2002,15 @@ StringView RenderListMarker::textWithoutSuffix() const
     return StringView { m_textWithSuffix }.left(m_textWithoutSuffixLength);
 }
 
-CSSCounterStyle* RenderListMarker::counterStyle() const
+RefPtr<CSSCounterStyle> RenderListMarker::counterStyle() const
+{
+    return document().counterStyleRegistry().resolvedCounterStyle(style().listStyleType().identifier);
+}
```

Comment:
This method should return nullptr for any `ListStyleType::Type` that isn't `CounterStyle` for accuracy.

Reply:
Also, separate, but `document().counterStyleRegistry().resolvedCounterStyle` should return a `Ref<>` instead of `RefPtr<>`